### PR TITLE
update documentation

### PR DESCRIPTION
**Pull request by zivillian**
First of all thanks for your documentation - it's the most complete I've found! While implementing an lvm parser in C# I found some additional sources and facts - this are my findings, which may be helpful.

### CODE REVIEW

File: documentation/Logical Volume Manager (LVM) format.asciidoc [hunk 1]
```diff
@@ -203,7 +205,7 @@ The offset, in bytes, relative from the start of the metadata area
 | 8 | 8 | | Data area size +
 Value in bytes
 | 16 | 4 | | Checksum +
-CRC-32 of [yellow-background]*TODO (metadata?)*
+CRC-32 for the data area
```
Comment (joachimmetz) [inline documentation/Logical Volume Manager (LVM) format.asciidoc:208]: which offset ranges? which polynomial ?

File: documentation/Logical Volume Manager (LVM) format.asciidoc [hunk 2]
```diff
@@ -291,8 +293,7 @@ Contains a list of strings. See section: <<status_flags,Status flags>>
 | flags | The flags +
 Contains a list of strings. See section: <<flags,Flags>>
 | extent_size | The size of an extent +
-The value contains the number of sectors +
-[yellow-background]*Note that the sector size is currently assumed to be 512 bytes.*
+The value contains the number of sectors
```
Comment (joachimmetz) [inline documentation/Logical Volume Manager (LVM) format.asciidoc:296]: please do not remove note regarding assumptions

Comment (joachimmetz) [inline documentation/Logical Volume Manager (LVM) format.asciidoc:296]: please do not remove note regarding assumptions

Comment (joachimmetz) [inline documentation/Logical Volume Manager (LVM) format.asciidoc:296]: please do not remove notes regarding assumptions (repeat this across your changes)

Comment (zivillian) [reply documentation/Logical Volume Manager (LVM) format.asciidoc:296]: It's not an assumption - the redhat documentation states it's defined as 512 bytes:
[The extent size (in units of sectors which are defined as 512 bytes) ](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/5/html/Logical_Volume_Manager_Administration/metadata_contents.html)

That's why I've added the sector size in the Layout section.

File: documentation/Logical Volume Manager (LVM) format.asciidoc (hunk 1 above)
Comment (zivillian) [reply documentation/Logical Volume Manager (LVM) format.asciidoc:208]: The CRC is calculated for the whole Data area (not the Raw location descriptor), defined by offset and size.
I've verified the checksum using the code from [pvdissect](http://www.syslinux.org/wiki/index.php?title=Development/LVM_support/pvdissect):

```
# LVM2.2.02.79:lib/misc/crc.c:_calc_crc_old()
INITIAL_CRC = 0xf597a6cf

def _calc_crc(buf, crc=INITIAL_CRC):
    crctab = [0x00000000, 0x1db71064, 0x3b6e20c8, 0x26d930ac,
              0x76dc4190, 0x6b6b51f4, 0x4db26158, 0x5005713c,
              0xedb88320, 0xf00f9344, 0xd6d6a3e8, 0xcb61b38c,
              0x9b64c2b0, 0x86d3d2d4, 0xa00ae278, 0xbdbdf21c]
    i = 0
    while i < len(buf):
        crc ^= ord(buf[i])
        crc = (crc >> 4) ^ crctab[crc & 0xf]
        crc = (crc >> 4) ^ crctab[crc & 0xf]
        i += 1
    return crc;
```

Comment (zivillian) [reply documentation/Logical Volume Manager (LVM) format.asciidoc:208]: This seems to be copied from the [NetBSD Ethernet CRC](https://github.com/ChaosData/LVMinfo.py/blob/master/lvminfo.py#L65).

**Comment by joachimmetz** [comment]
thanks I'll have a look at merging your changes as soon as time permits.

**Comment by Glimpse-006** [comment]
about the document ,there maybe wrong
------------------------------------------------------------
[cols="1,3",options="header"]
|===
| Value | Description
| id | Physical volume identifier (PV UUID) +
Contains an ASCII string in the following format: 9LBcEB-7PQT-GIlL-I0Kx-rtzr-ynju-SL983W
| device | The device filename +
Contains an ASCII string e.g. /dev/dm-0
| status | The status flags +
Contains a list of strings. See section: <<status_flags,Status flags>>
| dev_size | The physical volume size including non-usable space +
The value contains the number of sectors +
[yellow-background]*Note that the sector size is currently assumed to be 512 bytes.*
| pe_start | The start extent + /* correct by liubo(2018/12/8) The value contains the number of sectors + */
[yellow-background]*TODO: what is this value used for?*
| pe_count | The number of (allocated) extents in the physical volume
|===
------------------------------------------------------------

**Comment by Glimpse-006** [comment]
the document ,there maybe wrong,
the description about " pe_start"
---------------------------------------------------------
[cols="1,3",options="header"]
|===
| Value | Description
| id | Physical volume identifier (PV UUID) +
Contains an ASCII string in the following format: 9LBcEB-7PQT-GIlL-I0Kx-rtzr-ynju-SL983W
| device | The device filename +
Contains an ASCII string e.g. /dev/dm-0
| status | The status flags +
Contains a list of strings. See section: <<status_flags,Status flags>>
| dev_size | The physical volume size including non-usable space +
The value contains the number of sectors +
[yellow-background]*Note that the sector size is currently assumed to be 512 bytes.*
| pe_start | The start extent + /* correct by liubo(2018/12/8) The value contains the number of sectors + */
[yellow-background]*TODO: what is this value used for?*
| pe_count | The number of (allocated) extents in the physical volume
|===

**Comment by joachimmetz** [comment]
@Glimpse-006 first of all, thx for the report but please don't hijack open issue. Second could you be more explanatory, I have a hard time trying to parse what you are trying to tell me.

**Comment by Glimpse-006** [comment]
> 
> 
> @Glimpse-006 first of all, thx for the report but please don't hijack open issue. Second could you be more explanatory, I have a hard time trying to parse what you are trying to tell me.

(using translation tool -_-)
I would like to express that pe_start may be interpreted incorrectly in the document.(at lines of:335)
original:
| pe_start | The start extent +

Poor comprehension.
I thought you wanted to say that the value of  pe_startis an integer multiple of extent_size.
In my lvm disk, this value is the number of sectors and the offset of the partition on this device.
eg:
pv0 
{
	id = "r2tzsR-Vf9N-IpuD-EzJp-NQ4D-GfNa-2iBneZ"
	device = "/dev/sda"		
	dev_size = 62914560 (sec)
	pe_start = 384(sec) 
	pe_count = 7679
}
segment1 
{   
	start_extent = 0    
	extent_count = 2816     # 11 Gigabytes
	type = "striped"  
	stripe_count = 1        # linear
	stripes = [ "pv0", 3000  ]       
} 

segment1 offset in device /dev/sda is : pe_start+ 3000 + start_extent.

File: documentation/Logical Volume Manager (LVM) format.asciidoc (hunk 1 above)
Comment (joachimmetz) [reply documentation/Logical Volume Manager (LVM) format.asciidoc:208]: Finally had some time to look into this, its a weak CRC-32 (CRC without XOR with 0xffffffff)
using the polynominal 0xedb88320 and initial value 0xf597a6cf

**Comment by joachimmetz** [comment]
Changes integrated in https://github.com/libyal/libvslvm/commit/9e876c002da7dbcc26f0ca93e5377291e7becdb2, closing issue

**Pull request closed without merging**